lib/src/screens/cake_pay/cards/cake_pay_confirm_purchase_card_page.dart

```diff
@@ -263,6 +310,8 @@ class CakePayBuyCardDetailPage extends BasePage {
       } catch (_) {
         await cakePayPurchaseViewModel.cakePayService.logout();
       }
+
+      cakePayPurchaseViewModel.isPurchasing = false;
     }
```

Comment:
should set it to false either way, so that when he logs in, it would be false and he can tap the button
```suggestion
   }
   cakePayPurchaseViewModel.isPurchasing = false;
```